### PR TITLE
Add TrimFormatter for configurable string edge trimming

### PR DESCRIPTION
**Pull request by henriquemoody**
Allows precise control over trimming operations with support for left, right, or both sides and custom characters, using PHP's mb_trim, mb_ltrim, and mb_rtrim functions for proper multibyte-safe trimming.

Includes comprehensive tests covering all trim modes, custom characters, Unicode characters (CJK, emoji), special characters, multi-byte strings, and edge cases like empty strings and strings shorter than the characters to trim.

Assisted-by: OpenCode (GLM-4.7)

### CODE REVIEW

File: src/TrimFormatter.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use function preg_quote;
+use function preg_replace;
+use function sprintf;
+
+final readonly class TrimFormatter implements Formatter
+{
+    /** @param 'both'|'left'|'right' $side */
+    public function __construct(
+        private string $side = 'both',
+        private string $mask = " \t\n\r\0\x0B",
```
[the remark this answers is not in the document]
Comment (henriquemoody) [reply src/TrimFormatter.php]: I pick those from PHP's trim() default trimmed characters.

Do you mean we should __not__ merge it?

[the remark this answers is not in the document]
Comment (alganet) [reply src/TrimFormatter.php]: Also: https://www.php.net/mb_trim

[the remark this answers is not in the document]
Comment (henriquemoody) [reply src/TrimFormatter.php]: I've updated the commit message, the code, and the documentation, it should be much better now
